enhancements/ingress/gateway-api-with-cluster-ingress-operator.md

```diff
@@ -280,20 +272,98 @@ does the following:
 
 1. Install the Gateway API CRDs other than the gatewayclasses CRD, namely the gateways and httproutes CRDs.
 2. Create a Subscription to install OSSM.
-3. Create a ServiceMeshControlPlane to configure OSSM (what if it already exists?; see [Open Questions](#open-questions)).
-4. Create a LoadBalancer-type service (or OSSM may create it; see [Open Questions](#open-questions)).
-5. Create a DNSRecord for the service (see the following subsection).
+3. Create a ServiceMeshControlPlane to configure OSSM.
 
 The Ingress Operator effectively owns the Gateway API CRDs.  It installs
 versions of the CRDs that are compatible with the OSSM version that it installs,
 and if the CRDs are already installed, the operator may overwrite them.
 
-When OSSM is installed, it deploys the Istio control-plane and Envoy proxy.  The
-Ingress Operator configures the ServiceMeshControlPlane to enable Gateway API
-support, which means that Istio watches Gateway and HTTPRoute objects and
-configures Envoy accordingly.  Configuring a load balancer and DNS enables
-clients to connect through Envoy to backend applications as specified by the
-Gateway and HTTPRoute objects.
+When OSSM is installed, it deploys the Istio control-plane.  The Ingress
+Operator configures the ServiceMeshControlPlane to enable Gateway API support,
+which means that Istio watches Gateway and HTTPRoute objects and configures
+Envoy accordingly.  The Ingress Operator also configures the
+ServiceMeshControlPlane to enable Istio's "[Automated Deployment](https://istio.io/latest/docs/tasks/traffic-management/ingress/gateway-api/#automated-deployment)" feature,
+which means that Istio automatically creates an Envoy proxy deployment and a
+LoadBalancer-type service for each Gateway.  The Ingress Operator then
```

Comment:
Should we address the need for proxy protocol or other LB configuration options?

The default functionality is to not add any annotations  to the service, though you mentioned they can be carried over from the Gateway. So, in AWS, proxy protocol is disabled, and we get an ELB by default.

However, like we've talked about before, we don't preserve the client IP, which impacts debuggablity (since everything looks like it's coming from the ELB) and we can't do approve list deny list-type functions (though, that's a moot point because you can't do it anyways with GWAPI...).

I know there's a laundry list of things that we set as defaults for loadbalancers. Should we begin to address them here in a design? Will the ingress operator ever interact and override these annotations? How do we manage the philosophy of our  "opinionated defaults" when some of these options are now uncontrollable by us?

I'm guessing it's worth mentioning some of these configuration options, but for dev preview, I'd imagine we are just going to use the defaults and not have cluster-ingress-operator intervene.

Reply:
Yes, let's go with the defaults for dev preview.  I'll add a note.  

If the Ingress Operator does add these annotations, does the Service Mesh Operator ever overwrite them?  In any case, having two controllers updating the same object is an invitation for problems, so better we should find a way by which the Ingress Operator can override the Service Mesh Operator's template.

Reply:
https://github.com/openshift/enhancements/compare/2e06c5d3e457b776516f2e706c9567af34af82c9..a4236cc53936ad920e7aa63bd4a2957201a63aba#diff-7973eca3c2f112ece8f8585f4f4c6567fc1c1b4fa95be7f88a27ce08820d89f7 added the note.